### ISSUE TITLE
Crashalytics/Fabric upload fails with Shell command exited with exit status 1 instead of 0.

### ISSUE THREAD

**Issue by lakamsani** [fastlane/fastlane#4407]
### New Issue Checklist

- [x ] Updated fastlane to the latest version
- [ x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
Crashalytics/Fabric upload fails with the error:
[!] Shell command exited with exit status 1 instead of 0.
##### Complete output when running fastlane, including the stack trace and command used
command used: fastlane beta

output:

 ✗ fastlane beta
Ignoring xcodeproj-0.17.0 because its extensions are not built.  Try: gem pristine xcodeproj-0.17.0
[10:08:01]: -------------------------------------------------
[10:08:01]: --- Step: Verifying required fastlane version ---
[10:08:01]: -------------------------------------------------
[10:08:01]: fastlane version valid
[10:08:01]: ------------------------------
[10:08:01]: --- Step: default_platform ---
[10:08:01]: ------------------------------
[10:08:01]: Driving the lane 'ios beta' 🚀
[10:08:01]: ------------------------------------
[10:08:01]: --- Step: increment_build_number ---
[10:08:01]: ------------------------------------
[10:08:01]: $ cd /Users/me/Development/MyApp/MyApp-ios/iPhone && agvtool next-version -all && cd -
[10:08:01]: ▸ Setting version of project MyApp-iPhone to:
[10:08:01]: ▸ 6.
[10:08:01]: ▸ Also setting CFBundleVersion key (assuming it exists)
[10:08:01]: ▸ Updating CFBundleVersion in Info.plist(s)...
[10:08:01]: ▸ $(($buildNumber + 1))n#    /usr/libexec/PlistBuddy -c "Set :CFBundleVersion $buildNumber" "$INFOPLIST_FILE"n#fin
[10:08:01]: ▸ Cannot find "$(($buildNumber + 1))n#    /usr/libexec/PlistBuddy -c "Set :CFBundleVersion $buildNumber" "$INFOPLIST_FILE"n#fin"
[10:08:01]: ▸ Updated CFBundleVersion in "MyApp-iPhone.xcodeproj/../MyApp/Info.plist" to 6
[10:08:01]: ▸ Updated CFBundleVersion in "MyApp-iPhone.xcodeproj/../MyAppTests/Info.plist" to 6
[10:08:01]: ▸ Updated CFBundleVersion in "MyApp-iPhone.xcodeproj/../MyAppUITests/Info.plist" to 6
[10:08:01]: ▸ /Users/me/Development/MyApp/MyApp-ios
[10:08:02]: --------------------
[10:08:02]: --- Step: prompt ---
[10:08:02]: --------------------
[10:08:02]: Enter the change log:
[10:08:02]: Submit inputs using "END"
test
END
[10:08:06]: -----------------
[10:08:06]: --- Step: gym ---
[10:08:06]: -----------------
[10:08:06]: Using legacy build system - waiting for radar to be fixed: https://openradar.appspot.com/radar?id=4952000420642816
[10:08:06]: Installing provisioning profile...
[10:08:07]: xcrun xcodebuild -list -workspace 'MyApp.xcworkspace'
2016-04-28 10:08:07.286 xcodebuild[55130:435783] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2016-04-28 10:08:08.133 xcodebuild[55134:435817] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs

+---------------------------+----------------------------------------------------------------------+
|                                      Summary for gym 1.6.2                                       |
+---------------------------+----------------------------------------------------------------------+
| workspace                 | MyApp.xcworkspace                                                   |
| scheme                    | MyAppiPhoneDebug                                                    |
| output_directory          | archives                                                             |
| clean                     | true                                                                 |
| use_legacy_build_api      | true                                                                 |
| include_bitcode           | true                                                                 |
| provisioning_profile_path | /Users/me/Development/MyApp/MyApp-ios/MyApp_Adhoc_1.mobile... |
| destination               | generic/platform=iOS                                                 |
| output_name               | Pods_MyAppiPhone                                                    |
| silent                    | false                                                                |
| buildlog_path             | ~/Library/Logs/gym                                                   |
+---------------------------+----------------------------------------------------------------------+

[10:08:09]: $ set -o pipefail && xcodebuild -workspace 'MyApp.xcworkspace' -scheme 'MyAppiPhoneDebug' -destination 'generic/platform=iOS' -archivePath '/Users/me/Library/Developer/Xcode/Archives/2016-04-28/Pods_MyAppiPhone 2016-04-28 10.08.09.xcarchive' clean archive | tee /Users/me/Library/Logs/gym/Pods_MyAppiPhone-MyAppiPhoneDebug.log | xcpretty
[10:08:09]: ▸ 2016-04-28 10:08:09.889 xcodebuild[55153:435870] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Unity4XC.xcplugin' not present in DVTPlugInCompatibilityUUIDs
[10:08:11]: ▸ Cleaning Pods/EZSwiftExtensions [Debug]
[10:08:11]: ▸ Check Dependencies
[10:08:11]: ▸ Cleaning Pods/Locksmith [Debug]

====
--  logs with source files etc skipped---
=====

10:03:55]: ▸ Archive Succeeded
[10:03:56]: Successfully stored the archive. You can find it in the Xcode Organizer.
[10:03:56]: $ /usr/bin/xcrun /tmp/PackageApplication4Gym -v /Users/me/Library/Developer/Xcode/Archives/2016-04-28/Pods_MyAppiPhone\ 2016-04-28\ 10.02.52.xcarchive/Products/Applications/MyApp-iPhone.app -o '/Users/me/Library/Developer/Xcode/Archives/2016-04-28/Pods_MyAppiPhone.ipa' exportFormat ipa --embed '/Users/me/Development/MyApp/mapp-ios/MyApp_Adhoc_1.mobileprovision'
[10:03:59]: $ zip --recurse-paths '/Users/me/Library/Developer/Xcode/Archives/2016-04-28/Pods_MyAppiPhone.ipa' SwiftSupport > /dev/null
[10:04:01]: Successfully exported and signed the ipa file:
[10:04:01]: /Users/me/Development/MyApp/myapp-ios/archives/Pods_MyAppiPhone.ipa
[10:04:01]: -------------------------
[10:04:01]: --- Step: crashlytics ---
[10:04:01]: -------------------------
[10:04:01]: Uploading the build to Crashlytics Beta. Time for some ☕️.
[10:04:01]: Variable Dump:
[10:04:01]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :BUILD_NUMBER=>"5", :IPA_OUTPUT_PATH=>"/Users/me/Development/MyApp/myapp-ios/archives/Pods_MyAppiPhone.ipa", :XCODEBUILD_ARCHIVE=>"/Users/me/Library/Developer/Xcode/Archives/2016-04-28/Pods_MyAppiPhone 2016-04-28 10.02.52.xcarchive"}
[10:04:01]: Shell command exited with exit status 1 instead of 0.

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | increment_build_number              | 0           |
| 4    | prompt                              | 3           |
| 5    | gym                                 | 72          |
| 6    | crashlytics                         | 0           |
+------+-------------------------------------+-------------+

[10:04:01]: fastlane finished with errors

[!] Shell command exited with exit status 1 instead of 0.

```
INSERT OUTPUT HERE
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
fastlane_version "1.80.0"

default_platform :ios

platform :ios do
  before_all do
    ENV["SLACK_URL"] = "https://hooks.slack.com/services/xxx"
    ENV["CRASHLYTICS_API_TOKEN"] = "xyz"
    ENV["CRASHLYTICS_BUILD_SECRET"] = "abc"

#    cocoapods
    
  end

  desc "Runs all the tests"
  lane :test do
    scan
    clean_build_artifacts
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    #xctool
#    ensure_git_status_clean
#    ensure_git_branch(branch: 'master')
#    git_pull
#    push_to_git_remote
    # Ask the user of fastlane to enter a changelog
    increment_build_number(xcodeproj: './iPhone/MyApp-iPhone.xcodeproj')
    change_log = prompt(text: "Enter the change log: ",
                     multi_line_end_keyword: "END")
                     
    gym(
      workspace: 'MyApp.xcworkspace',
      scheme: 'MyAppDebug',
      output_directory: "archives",
      clean: true, use_legacy_build_api:true, include_bitcode:true
    ) 
    crashlytics(
        api_token: ENV["CRASHLYTICS_API_TOKEN"],
        build_secret: ENV["CRASHLYTICS_BUILD_SECRET"],
        notes: change_log
     )
    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :appstore do
    # match(type: "appstore")
    # snapshot
    gym # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    #slack(
    #  message: "Successfully deployed new App Update."
    #)
  end

  error do |lane, exception|
     #slack(
     #  message: exception.message,
     #  success: false
     #)
  end
end

```

### Environment

fastlane version (run `fastlane -v`):
✗ fastlane -version
Ignoring xcodeproj-0.17.0 because its extensions are not built.  Try: gem pristine xcodeproj-0.17.0
fastlane 1.83.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? No

**Comment by etolstoy** [comment]
+1, the same problem.

**Comment by TKBurner** [comment]
Hi @lakamsani. Can you run your lane with the --verbose flag and share the output of the Crashlytics action? I want to take a look if this is a network issue or something else. Thanks!

**Comment by lakamsani** [comment]
Hi @TKBurner thanks for the flag. That produced some extra output that might help. 

==
INFO [2016-04-29 09:41:45.31]: Successfully exported and signed the ipa file:
INFO [2016-04-29 09:41:45.31]: /Users/me/Development/MyApp/MyApp-ios/archives/Pods_MyAppiPhone.ipa
INFO [2016-04-29 09:41:45.31]: -------------------------
INFO [2016-04-29 09:41:45.31]: --- Step: crashlytics ---
INFO [2016-04-29 09:41:45.31]: -------------------------
INFO [2016-04-29 09:41:45.34]: Uploading the build to Crashlytics Beta. Time for some ☕️.
DEBUG [2016-04-29 09:41:45.34]: ./Pods/Crashlytics/iOS/Crashlytics.framework/submit b5aed52852546423058dbffb60da4e25732385e b094090903db381132aa64b20c93df7008b1b4a03856f7df021fcb2b602c544a -ipaPath '/Users/me/Development/MyApp/MyApp-ios/archives/Pods_MyAppiPhone.ipa' -notifications YES -debug NO
WARN [2016-04-29 09:41:45.36]: Variable Dump:
INFO [2016-04-29 09:41:45.36]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :BUILD_NUMBER=>"10", :IPA_OUTPUT_PATH=>"/Users/me/Development/MyApp/MyApp-ios/archives/Pods_MyAppiPhone.ipa", :XCODEBUILD_ARCHIVE=>"/Users/me/Library/Developer/Xcode/Archives/2016-04-29/Pods_MyAppiPhone 2016-04-29 09.40.26.xcarchive"}
ERROR [2016-04-29 09:41:45.36]: Shell command exited with exit status 1 instead of 0.
INFO [2016-04-29 09:41:45.37]: Successfully generated documentation to path '/Users/me/Development/MyApp/MyApp-ios/fastlane/README.md'

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | increment_build_number              | 1           |
| 4    | gym                                 | 82          |
| 5    | crashlytics                         | 0           |
+------+-------------------------------------+-------------+

ERROR [2016-04-29 09:41:45.37]: fastlane finished with errors
/Users/me/Software/ruby/gems/fastlane_core-0.42.1/lib/fastlane_core/ui/interface.rb:139:in `user_error!': [!] Shell command exited with exit status 1 instead of 0. (FastlaneCore::Interface::FastlaneError)
	from /Users/me/Software/ruby/gems/fastlane_core-0.42.1/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/helper/sh_helper.rb:47:in `sh_control_output'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/helper/sh_helper.rb:9:in `sh'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/actions/crashlytics.rb:32:in `run'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:143:in `block (2 levels) in execute_action'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:129:in `block in execute_action'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:128:in `chdir'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:128:in `execute_action'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:54:in `block (2 levels) in parsing_binding'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/lane.rb:36:in `call'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/lane.rb:36:in `call'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:52:in `block in execute'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:46:in `chdir'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/runner.rb:46:in `execute'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/bin/fastlane:38:in `block (2 levels) in run'
	from /Users/me/Software/ruby/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Users/me/Software/ruby/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Users/me/Software/ruby/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Users/me/Software/ruby/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Users/me/Software/ruby/gems/fastlane_core-0.42.1/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Users/me/Software/ruby/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/bin/fastlane:168:in `run'
	from /Users/me/Software/ruby/gems/fastlane-1.83.0/bin/fastlane:174:in `<top (required)>'
	from /Users/me/Software/ruby/bin/fastlane:23:in `load'
	from /Users/me/Software/ruby/bin/fastlane:23:in `<main>'

**Comment by TKBurner** [comment]
Hi @lakamsani! Thanks for following up. Can you change your notes parameter to 'notes_path'? Let me know if that clears things up on your end! Thanks!

**Comment by lakamsani** [comment]
@TKBurner  when I ran  with the --verbose option to generate the error details in my last post I actually had the notes field commented out like this:

#notes: change_log

I tried with the following and still the same error. 

notes_path: "/tmp/notes.txt"

**Comment by TKBurner** [comment]
@lakamsani  Hi, Can you try manually setting the crashlytics_path? I want to see if this clears things up for you. Thanks for working with me 👍

**Comment by lakamsani** [comment]
@TKBurner  added this line:

  crashlytics_path: "./Crashlytics.framework/"

The Crashlytics framework is in the same(current) directory where I run the fastlane beta command. 'ls' shows this:

===
✗ ls Crashlytics.framework
Crashlytics* Headers/     Info.plist   Modules/     run*         submit*      uploadDSYM*
✗
====

But I still get the same error. I tried upgrading to fastlane 1.85 .0 without any change in the error.

**Comment by etolstoy** [comment]
My problem is solved - it just began to wok normally. Thanks everyone.

**Comment by sweetmandm** [comment]
@lakamsani In my experience the fastlane crashlytics command will not show specific error details, even with --verbose. you might try running the Crashlytics command manually. The `submit` command will live in a different place based on how you've installed Crashlytis (Pods vs. drag-n-drop), For me, my command looked like:
```
./Crashlytics.framework/submit <api-token> <build-secret> -ipaPath './myApp.ipa' -groupAliases 'my-internal-group' -notifications 'YES'
```
This will print out the descriptive crashlytics errors, and when I saw this output the problems were easily resolved by modifying my Fastfile and my project settings at fabric.io.

**Comment by sweetmandm** [comment]
@lakamsani the error output should be improved by https://github.com/fastlane/fastlane/pull/4526, which is in fastlane-1.86.0. If you update fastlane to the new version and try the command again, you might get some more descriptive error information.

**Comment by lakamsani** [comment]
@sweetmandm thanks for your feedback. Your comment made me look closely at the debug logs and instead of focussing the ruby exception stack trace I looked a few lines above and noticed this. 

===
/Pods/Crashlytics/iOS/Crashlytics.framework/submit <key> <secret> -ipaPath '/Users/me/Development/MyApp/myapp-ios/archives/Pods_MyAppiPhone.ipa' -notifications YES -debug NO
2016-05-05 12:56:40.998 submit Crashlytics: Crashlytics.framework/submit 1.3.5 (17)
2016-05-05 12:56:41.005 submit Crashlytics: First argument must be API Key, not <key>
===

I removed the actual key from the log above but comparing it with the Fabric dashboard I noticed that I missed the first letter of the key when pasting it into my Fastfile's ENV["CRASHLYTICS_API_TOKEN"] setting. After I fixed that the upload started working. I also updated to 1.86.0. We can close this now.
